src/main/java/uk/gov/pay/adminusers/service/ServiceUserRemover.java

```diff
@@ -29,13 +29,14 @@ public ServiceUserRemover(UserDao userDao, ServiceRoleDao serviceRoleDao) {
     public void remove(String userExternalId, String removerExternalId, String serviceExternalId) {
         LOGGER.info("User remove from service requested - serviceId={}, removerId={}, userId={}", serviceExternalId, removerExternalId, userExternalId);
         userDao.findByExternalId(userExternalId)
-                .flatMap(userEntity -> userEntity.getServicesRole(serviceExternalId)
+                .map(userEntity -> userEntity.getServicesRole(serviceExternalId)
```

Comment:
Is there a way in which we can get rid of some of this nesting. Maybe assigning to optional variables and throwing if empty? My Java knowledge isn't up to thinking of a way to do it but I think if it could be done it would perhaps be clearer.